projects/ngx-translate/core/src/lib/translate.service.ts

```diff
@@ -238,20 +238,27 @@ export class TranslateService {
    */
   public getTranslation(lang: string): Observable<any> {
     this.pending = true;
-    const loadingTranslations = this.currentLoader.getTranslation(lang).pipe(share());
-    this.loadingTranslations = loadingTranslations.pipe(
+    const loadingTranslations = this.currentLoader.getTranslation(lang).pipe(
+      shareReplay(1),
       take(1),
+    );
+
+    this.loadingTranslations = loadingTranslations.pipe(
       map((res: Object) => this.compiler.compileTranslations(res, lang)),
-      share()
+      shareReplay(1),
+      take(1),
```

Comment:
This block is a little refactored because it was actually making multiple subscriptions when `this.currentLoader.getTranslation(lang)` was synchronous. In such case `share()` had no effect. There's a test https://github.com/ngx-translate/core/pull/1137/files#diff-3642e241860a02564009cf341f922874L396 that was supposed to test this scenario but it tests only async behavior. I added one more test down here https://github.com/ngx-translate/core/pull/1137/files#diff-3642e241860a02564009cf341f922874R410 that covers also sync `getTranslation()`